src/pow.cpp

```diff
@@ -47,8 +43,8 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
         return pindexLast->nBits;
     }
 
-    // Go back by what we want to be 14 days worth of blocks
-    int nHeightFirst = pindexLast->nHeight - (params.DifficultyAdjustmentInterval()-1);
+    // Calculate the diff for the next block based on the block time in the last 10 blocks
+    int nHeightFirst = pindexLast->nHeight - NUM_OF_BLOCKS_DIFF_ADJ;
```

Comment:
Maybe nHeightFirst = min(0, nHeightFirst)?
Also, I believe the new consensus rule should be only applied to blocks after the fork.